pkg/tsdb/cloudwatch/metric_find_query_test.go

```diff
@@ -499,3 +499,49 @@ func TestQuery_ResourceARNs(t *testing.T) {
 		}, resp)
 	})
 }
+
+func TestQuery_ListMetricsPagination(t *testing.T) {
+	origNewCWClient := NewCWClient
+	t.Cleanup(func() {
+		NewCWClient = origNewCWClient
+	})
+
+	var client FakeCWClient
+
+	NewCWClient = func(sess *session.Session) cloudwatchiface.CloudWatchAPI {
+		return client
+	}
+
+	metrics := []*cloudwatch.Metric{
+		{MetricName: aws.String("Test_MetricName1")},
+		{MetricName: aws.String("Test_MetricName2")},
+		{MetricName: aws.String("Test_MetricName3")},
+		{MetricName: aws.String("Test_MetricName4")},
+		{MetricName: aws.String("Test_MetricName5")},
+		{MetricName: aws.String("Test_MetricName6")},
+		{MetricName: aws.String("Test_MetricName7")},
+		{MetricName: aws.String("Test_MetricName8")},
+		{MetricName: aws.String("Test_MetricName9")},
+		{MetricName: aws.String("Test_MetricName10")},
+	}
+
+	t.Run("List Metrics and page limit is reached", func(t *testing.T) {
+		client = FakeCWClient{Metrics: metrics, MetricsPerPage: 2}
+		executor := newExecutor(nil, &awsSettings{ListMetricsPageLimit: 3})
+		executor.DataSource = fakeDataSource()
+		response, err := executor.listMetrics("default", &cloudwatch.ListMetricsInput{})
+		require.NoError(t, err)
+
+		assert.Equal(t, 6, len(response))
```

Comment:
That's fine, I just usually prefer when I have only numbers (like 6 here), to define it since we may forget what this number stands for in the future. Maybe something like 

```
expectedMetrics := client.MetricsPerPage * executor.settings.ListMetricsPageLimit
assert.Equal(t, expectedMetrics, len(response))
```
Feel free to disagree though, it's a personal preference :)

Reply:
Yeah I agree